modules/mesos-cluster/main.tf

```diff
@@ -12,6 +12,35 @@ variable "slave_as_max_size" {}
 variable "slave_as_desired_capacity" {}
 variable "slave_as_min_size" {}
 
+data "aws_iam_policy_document" "mesos-${var.environment}-assume-role-policy" {
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Principal": {
+                "Service": "ec2.amazonaws.com"
+            },
+            "Action": "sts:AssumeRole"
+        }
+    ]
+}
+
+resource "aws_iam_role" "mesos-${var.environment}-role" {
+    name = "Mesos-${Var.Environment}-Role"
+    assume_role_policy = "${data.aws_iam_policy_document.mesos-${var.environment}-assume-role-policy.json}"
+}
+
+resource "aws_iam_role_policy_attachment" "mesos-${var.environment}-access-policy" {
+    role = "${aws_iam_role.mesos-${var.environment}-role.name}"
+    policy_arn = "arn:aws:iam::484535289196:policy/SnsMozdefLogsFullAccess"
+}
+
+resource "aws_iam_instance_profile" "mesos-${var.environment}-profile" {
```

Comment:
You'll also need to add the instance profile to the launch configs of the master/slaves